dev/ci/README.md

```diff
@@ -183,3 +183,16 @@ but if you wish to save more time you can skip the job by setting
 This means you will need to change its value when the Docker image
 needs to be updated. You can do so for a single pipeline by starting
 it through the web interface.
+
+### Artifacts
```

Comment:
There's already some information about GitLab artifacts at the beginning on the GitLab section, so I would prefer if it could be extended rather than adding a new section here. At least, gather everything together in just one place.